### PR TITLE
[4.12] OCPBUGS-6789: backport feature gate honoring for PSa label syncer

### PR DESCRIPTION
**Pull request by stlaz**
this is a backport of https://github.com/openshift/cluster-policy-controller/pull/89

/cc @deads2k @ibihim

### CODE REVIEW

**Comment by deads2k** [comment]
/approve
/assign @ibihim 

for lgtm

File: pkg/cmd/controller/psalabelsyncer.go [hunk 1]
```diff
@@ -13,19 +14,40 @@ func runPodSecurityAdmissionLabelSynchronizationController(ctx context.Context,
...
-		controllerCtx.KubernetesInformers.Core().V1().Namespaces(),
-		controllerCtx.KubernetesInformers.Rbac().V1(),
-		controllerCtx.KubernetesInformers.Core().V1().ServiceAccounts(),
-		controllerCtx.SecurityInformers.Security().V1().SecurityContextConstraints(),
-		controllerCtx.EventRecorder.ForComponent("podsecurity-admission-label-sync-controller"),
-	)
+	featureGates := sets.NewString(controllerCtx.OpenshiftControllerConfig.FeatureGates...)
+	switch {
+	case featureGates.Has("OpenShiftPodSecurityAdmission=true"):
+		// if explicitly off, disable
```
Comment (ibihim) [inline pkg/cmd/controller/psalabelsyncer.go:20]: This comment is slightly confusing to me. In this case we are enabling the enforcement.

**Review by ibihim — APPROVED** [review]
/lgtm

**Pull request merged**